### ISSUE TITLE
Server does not build under Linux

### ISSUE THREAD

**Issue by theCapypara** [Saghetti0/OpenPMD#1]
Hey,
thought I'd test building under Linux for you :)
Building the client works fine, but the server not, it seems to try to find an exe in the end?

```
Checking files for target server
File src/server/net_main.c modified since last build, recompiling
gcc -c -Wall -std=c99 src/server/net_main.c -o obj/server/src/server/net_main.o
src/server/net_main.c: In function ‘net_serve’:
src/server/net_main.c:85:88: warning: pointer targets in passing argument 3 of ‘accept’ differ in signedness [-Wpointer-sign]
   85 |     int client_socket_fp = accept(net_server_socket_fd,(struct sockaddr *)&client_addr,&client_addr_len);
      |                                                                                        ^~~~~~~~~~~~~~~~
      |                                                                                        |
      |                                                                                        int *
In file included from /usr/include/netinet/in.h:23,
                 from /usr/include/netdb.h:27,
                 from src/server/net_conn_info.h:9,
                 from src/server/net_main.h:8,
                 from src/server/net_main.c:9:
/usr/include/sys/socket.h:233:28: note: expected ‘socklen_t * restrict’ {aka ‘unsigned int * restrict’} but argument is of type ‘int *’
  233 |      socklen_t *__restrict __addr_len);
      |      ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
File src/server/net_conn_handler.c modified since last build, recompiling
gcc -c -Wall -std=c99 src/server/net_conn_handler.c -o obj/server/src/server/net_conn_handler.o
File src/server/main.c modified since last build, recompiling
gcc -c -Wall -std=c99 src/server/main.c -o obj/server/src/server/main.o
Linking...
gcc obj/server/src/server/main.o obj/server/src/server/net_conn_handler.o obj/server/src/server/net_main.o -llua -lpthread -o bin/server/openpmd_server.exe
/usr/bin/ld: cannot open output file bin/server/openpmd_server.exe: No such file or directory
collect2: error: ld returned 1 exit status
Compiler returned code 256, exiting!
```

**Comment by waltermin** [comment]
Fixed with commit f4e2280986df643b15dcf95841f492dc3bbb8897

**Comment by waltermin** [comment]
This was caused due to the /bin/server/ directory not being automatically created, therefore causing ld to be unable write the output file. The build_config.json was also causing it to attempt to output in .EXE format.

The warnings while compiling are due to using an int instead of a socklen_t type, because most of the testing was done on Windows using winsock. However this doesn't cause any issues. I will fix this in a later commit.
